File: kubik/kubik-server/src/main/java/com/cspinformatique/kubik/server/domain/warehouse/service/ProductInventoryService.java
package com.cspinformatique.kubik.server.domain.warehouse.service;

import java.text.DecimalFormat;
import java.util.Date;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.cspinformatique.kubik.server.domain.warehouse.model.InventoryExtract;
import com.cspinformatique.kubik.server.model.product.Product;
import com.cspinformatique.kubik.server.model.warehouse.ProductInventory;

public interface ProductInventoryService {
	double calculateProductOnHandUntil(int productId, Date until);
	
	void deleteByProduct(Product product);

	List<ProductInventory> findAll();

	Page<ProductInventory> findAll(Pageable pageable);

	ProductInventory findByProduct(Product product);

	List<Integer> findProductIdWithInventory();

	InventoryExtract generateInventoryExtraction(String separator, DecimalFormat decimalFormat);
	
	InventoryExtract generateInventoryExtraction(String separator, DecimalFormat decimalFormat, Date until);

	void updateInventory(Product product);

	void updateInventory(ProductInventory productInventory);

	void updateInventories();

	ProductInventory save(ProductInventory productInventory);
}

File: kubik/kubik-server/src/main/webapp/public/js/warehouse/product-inventory-service.js
(function(){
	var PRODUCT_INVENTORY = "/productInventory";
	
	angular
		.module("Kubik")
		.factory("productInventoryService", ProductInventoryService);
	
	function ProductInventoryService($http){
		return {
			updateProductInventory : updateProductInventory
		};
		
		function updateProductInventory(ean13){
			return $http
				.get(PRODUCT_INVENTORY + "/product/" + ean13 + "?updateInventory")
				.then(updateProductInventorySuccess);
			
			function updateProductInventorySuccess(response){
				return response.data;
			}
		}
	}
})();